### DISCUSSION TITLE
Get the namespaces

### DISCUSSION THREAD

**Discussion by hfhbd** [pdvrieze/xmlutil#201] [Q&A] [answered]
I am building a XSD/WSDL/Swagger/OpenAPI to Kotlin code generator and I need to get the referenced namespace:
```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<wsdl:definitions
        xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/"
        xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/"
        xmlns:xsd="http://www.w3.org/2001/XMLSchema"
        name="Foo"
        xmlns:wsa="http://www.w3.org/2005/08/addressing"
        xmlns:wsp="http://www.w3.org/ns/ws-policy"
        xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"
        targetNamespace="http://www.example.com"
        xmlns:css="http://www.example.com/css"
>
    <wsdl:message name="fooMessage">
        <wsdl:part element="css:fooMessage" name="parameters"/>
    </wsdl:message>
</wsdl:definitions>
```
Is there any api to get the namespaces or the namespace for a specific prefix, like `css => http://www.example.com/css`, on the JVM?
I use the kotlinx-serialization module, version `0.86.3`.

**Comment by hfhbd** [comment, marked as answer]
Wow, this was easier than expected: `XMLReader` exposes the method: `getNamespaceURI(prefix)`

**Reply by pdvrieze** [reply]
`XmlReader` is intended to be a full (pull) parser implementation and able to be used without ever getting to the serialization elements (or library - core is sufficient).
